trieste/acquisition/optimizer.py

```diff
@@ -413,9 +529,16 @@ def batchify(
     if batch_size <= 0:
         raise ValueError(f"batch_size must be positive, got {batch_size}")
 
-    def optimizer(search_space: SP, f: AcquisitionFunction) -> TensorType:
+    def optimizer(
+        search_space: SP, f: Union[AcquisitionFunction, Tuple[AcquisitionFunction, int]]
+    ) -> TensorType:
         expanded_search_space = search_space ** batch_size  # points have shape [B * D]
 
+        if isinstance(f, tuple):
+            raise ValueError(
+                "batchify_joint cannot be applied to a vectorized acquisition function"
+            )
+
```

Comment:
The type error here is because f is being called inside a closure: by the time target_func_with_vectorized_inputs gets called f may have been assigned a new value. I think you can work around this by adding the following after the isinstance, and using `af` rather than `f` in target_func_with_vectorized_inputs:

```suggestion
af : AcquisitionFunction = f
```